src/shims/intrinsics.rs

```diff
@@ -417,6 +417,16 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
             "atomic_xsub_acqrel" => this.atomic_op(args, dest, BinOp::Sub, false, AtomicRwOp::AcqRel)?,
             "atomic_xsub_relaxed" => this.atomic_op(args, dest, BinOp::Sub, false, AtomicRwOp::Relaxed)?,
 
+            "atomic_min" => this.atomic_min_max(args, dest, true, AtomicRwOp::SeqCst)?,
```

Comment:
Is there anything fundamentally different between `atomic_op` and `atomic_min_max`? Would it make sense to merge them into a single function? As we just established, they should be basically the same in terms of synchronization behavior, right?

Reply:
AFAIK The `atomic_op` appears to take its output from `binary_op` with overflow which I think means that for `LT` / `GT` would result in new becoming the condition of the branch (e.g. `1`) rather than being max or min

Reply:
Sorry if I am missing something I am obviously super new to the miri codebase

Reply:
Right, so currently it uses `mir::BinOp` to determine the `atomic_op`. But we could instead use an enum like
```rust
enum AtomicOp {
  MirOp(mir::BinOp),
  Max,
  Min
}
```
and then we should be able to share much more code -- maybe?

> Sorry if I am missing something I am obviously super new to the miri codebase

Don't worry, these are good questions. :)

Reply:
I am ok with that, I was under the impression the style was to keep towards `mir` more directly. I will play with that a bit when I figure out the test case and my test failures.

I guess the other option is to match on `mir::BinOp::{Lt, Gt}` too which removes the need for a wrapper and punts the specifics to `atomic_op` I will also play with that

Reply:
`Lt` is for less-than though, and this is not "atomic less-than", it is "atomic min". Using `Lt` for this seems rather hacky.

> I was under the impression the style was to keep towards mir more directly.

That's just what worked well so far.^^ But given that all the fetch_ operations behave very similarly, I feel we really should share as much code between them as we can.